File: mobile/lib/domain/models/inventory.dart
class InventoryItem {
  final String id;
  final String? categoryId;
  final String name;
  final String? description;
  final String? brand;
  final String? model;
  final String? serialNumber;
  final String? purchaseDate;
  final String? purchasePrice;
  final String? currentValue;
  final String condition;
  final String? location;
  final String? warrantyExpiry;
  final List<String>? tags;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  InventoryItem({
    required this.id,
    this.categoryId,
    required this.name,
    this.description,
    this.brand,
    this.model,
    this.serialNumber,
    this.purchaseDate,
    this.purchasePrice,
    this.currentValue,
    this.condition = 'good',
    this.location,
    this.warrantyExpiry,
    this.tags,
    this.isActive = true,
    required this.createdAt,
    required this.updatedAt,
  });

  factory InventoryItem.fromJson(Map<String, dynamic> json) {
    return InventoryItem(
      id: json['id'],
      categoryId: json['categoryId'],
      name: json['name'],
      description: json['description'],
      brand: json['brand'],
      model: json['model'],
      serialNumber: json['serialNumber'],
      purchaseDate: json['purchaseDate'],
      purchasePrice: json['purchasePrice'],
      currentValue: json['currentValue'],
      condition: json['condition'] ?? 'good',
      location: json['location'],
      warrantyExpiry: json['warrantyExpiry'],
      tags: json['tags'] != null ? List<String>.from(json['tags']) : null,
      isActive: json['isActive'] ?? true,
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}

File: backend/src/db/schema/inventory.ts
import { pgTable, uuid, varchar, text, timestamp, decimal, boolean, date, index } from "drizzle-orm/pg-core";
import { users } from "./users";
import { itemConditionEnum } from "./enums";

export const inventoryCategories = pgTable("inventory_categories", {
    id: uuid("id").primaryKey().defaultRandom(),
    userId: uuid("user_id").references(() => users.id, { onDelete: "cascade" }).notNull(),
    name: varchar("name", { length: 100 }).notNull(),
    description: text("description"),
    icon: varchar("icon", { length: 50 }),
    createdAt: timestamp("created_at", { withTimezone: true }).defaultNow(),
}, (t) => [
    index("inventory_categories_user_id_idx").on(t.userId),
]);

export const inventoryItems = pgTable("inventory_items", {
    id: uuid("id").primaryKey().defaultRandom(),
    userId: uuid("user_id").references(() => users.id, { onDelete: "cascade" }).notNull(),
    categoryId: uuid("category_id").references(() => inventoryCategories.id),
    name: varchar("name", { length: 255 }).notNull(),
    description: text("description"),
    brand: varchar("brand", { length: 100 }),
    model: varchar("model", { length: 100 }),
    serialNumber: varchar("serial_number", { length: 100 }),
    purchaseDate: date("purchase_date"),
    purchasePrice: decimal("purchase_price", { precision: 15, scale: 2 }),
    currentValue: decimal("current_value", { precision: 15, scale: 2 }),
    currency: varchar("currency", { length: 10 }).default("IDR"),
    condition: itemConditionEnum("condition").default("good"),
    location: varchar("location", { length: 100 }),
    warrantyExpiry: date("warranty_expiry"),
    receiptUrl: text("receipt_url"),
    photoUrls: text("photo_urls").array(),
    tags: varchar("tags", { length: 50 }).array(),
    isActive: boolean("is_active").default(true),
    createdAt: timestamp("created_at", { withTimezone: true }).defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).defaultNow(),
}, (t) => [
    index("inventory_items_user_id_idx").on(t.userId),
]);